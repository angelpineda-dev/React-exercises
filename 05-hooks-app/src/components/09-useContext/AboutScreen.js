import { useContext } from "react";
import { UserContext } from "./UserContext";

const AboutScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };
  return (
    <div>
      <h2>About</h2>
      <button className="btn btn-outline-primary" onClick={handleClick}>
        Logout
      </button>
    </div>
  );
};

export default AboutScreen;
