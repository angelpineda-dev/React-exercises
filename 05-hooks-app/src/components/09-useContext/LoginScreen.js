import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const userData = {
    id: 123,
    name: "Angel Pineda",
  };

  return (
    <div>
      <h2>Login</h2>
      <button
        className="btn btn-outline-primary"
        onClick={() => setUser(userData)}
      >
        Login
      </button>
    </div>
  );
};
