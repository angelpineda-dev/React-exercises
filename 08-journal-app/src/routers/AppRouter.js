import { getAuth, onAuthStateChanged } from "@firebase/auth";
import { useState } from "react";
import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import { login } from "../actions/auth";
import JournalScreen from "../components/journal/JournalScreen";
import AuthRouter from "./AuthRouter";
import PrivateRoute from "./PrivateRoute";
import PublicRoute from "./PublicRoute";

const AppRouter = () => {
  const [checking, setChecking] = useState(true);
  const [isLogged, setIsLogged] = useState(false);
  const dispatch = useDispatch();

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setIsLogged(true);
      } else {
        setIsLogged(false);
      }

      setChecking(false);
    });
  }, [setIsLogged, dispatch]);

  if (checking) {
    return <h1>Cargando...</h1>;
  }

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            isLogged={isLogged}
            component={AuthRouter}
            path="/auth"
          />

          {/* <Route path="/" component={JournalScreen} exact /> */}

          <PrivateRoute
            isLogged={isLogged}
            component={JournalScreen}
            path="/"
          />

          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
