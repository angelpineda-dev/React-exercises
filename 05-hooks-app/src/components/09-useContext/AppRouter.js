import { HashRouter, Switch, Route, Link } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import NavBar from "./NavBar";
const AppRouter = () => {
  return (
    <HashRouter>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route exact path="/about" component={AboutScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Route
            path="*"
            children={
              <div>
                <h1>Error 404 - Not Found</h1>
                <Link to="/">Regresar</Link>
              </div>
            }
          ></Route>
        </Switch>
      </div>
    </HashRouter>
  );
};

export default AppRouter;
