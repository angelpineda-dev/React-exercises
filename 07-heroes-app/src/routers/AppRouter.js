import { useContext } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { AuthContext } from "../auth/AuthContext";
import LoginScreen from "../components/login/LoginScreen";
import DashboardRoutes from "./DashboardRoutes";
import PrivateRoute from "./PrivateRoute";

export const AppRouter = () => {
  const { user } = useContext(AuthContext);

  return (
    <Router>
      <Switch>
        <Route exact path="/login" component={LoginScreen} />
        <PrivateRoute
          isAuth={user.logged}
          path="/"
          component={DashboardRoutes}
        />
      </Switch>
    </Router>
  );
};
