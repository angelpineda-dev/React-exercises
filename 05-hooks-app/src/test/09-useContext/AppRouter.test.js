import React from "react";
import AppRouter from "../../components/09-useContext/AppRouter";
import { mount } from "enzyme";
import { UserContext } from "../../components/09-useContext/UserContext";

describe("Pruebas en <AppRouter/>", () => {
  const user = {
    id: 1,
    name: "Angel",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test("debe mostrarse correctamtente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
