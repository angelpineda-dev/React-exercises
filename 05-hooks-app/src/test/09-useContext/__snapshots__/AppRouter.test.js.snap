// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> debe mostrarse correctamtente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="#/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="#/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="#/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div>
    <h2>
      Home
    </h2>
    <pre>
      {
   "id": 1,
   "name": "Angel"
}
    </pre>
  </div>
</div>
`;
